src/src/hooks/SelfServiceApi.js

```diff
@@ -34,31 +36,33 @@ export function useSelfServiceRequest() {
                 const newData = await httpResponse.json();
                 setResponseData(newData);
             } else {
-                setErrorMessage("Oh no! We had an issue while retrieving data from the api. Please reload the page.");
-                // if (httpResponse.headers.get("Content-Type") === "application/problem+json") {
-                //     const { detail } = await httpResponse.json();
-                //     setErrorMessage(detail);
-                // } else {
-                //     setErrorMessage("Oh no! We had an issue while retrieving data from the api. Please reload the page.");
-                // }
+                triggerError(NewErrorTriggerRequestBuilder()
+                  .setHttpResponse(httpResponse)
+                  // .setHandler((params) => {
+                  //   if (params.httpResponse.headers.get("Content-Type") === "application/problem+json") {
+                  //       const { detail } = await httpResponse.json();
+                  //       setErrorMessage(detail);
+                  //   } else {
+                  //       setErrorMessage("Oh no! We had an issue while retrieving data from the api. Please reload the page.");
+                  //   }
+                  // })
+                  .build());
+
             }
         } catch (error) {
-            setErrorMessage(error.message);
+            triggerError(NewErrorTriggerRequestBuilder()
```

Comment:
You were asking for naming right? Could we name it NewTriggerErrorRequestBuilder to keep the same word ordering (triggerError vs errorTrigger).

Do we need the word Request in the name? Or can we omit it or find a better word.

Reply:
Thought about it for a bit, updated it to be ErrorContext instead of ErrorTriggerRequest (https://github.com/dfds/selfservice-portal/pull/25/commits/62b93bd52ae0472919825691ab39f757d92bb898). Thoughts?

Reply:
sounds good